### PR TITLE
Fix slcan w IsoTPSoftSocket

### PR DESCRIPTION
**Pull request by BenGardiner**
( I worked on this one with copilot / Claude 4.6 and also some gemini-cli over in https://github.com/BenGardiner/scapy/pull/2/ )

In addition to the change introduced in #4929 , this PR also introduces testcases to trigger the isotp soft socket regression with sr1() timeout on an MF response to a SF request that I have been experiencing . And then it fixes it by:
* removing bus filters for slcan interfaces, relying instead on the mux to filter
* optimizing time under lock and data receive latency in the python-can mux
* setting timeouts in isotp soft socket that are needed for slow interfaces such as slcan
* don't call select() (v slow on slcan) if the internal state will do

Verification:
- without the fixes the unit tests fail on slow interfaces
- with the fixes all unit tests pass
- Manual Testing (on the same setup that causes the failure noted in #4838, those table rows included for reference) 

| ver | driver | bg traffic | **sr1() threaded=?** | ISOTP DNE result | RDID SF req + MF resp result |
|-----|--------|-----------|-------------------|--------------------|-------------------------------|
|2.6.0-rc2| candle | YES | False | timeouts ✅ | data ✅ | 
|2.6.0-rc2| candle | YES | **True** | **hang** ❌ |  **data** ✅ |
|2.6.0-rc2| slcan | YES | False | timeouts ✅ | timeouts ❌ | 
|2.6.0-rc2| slcan | YES | **True** | **hang** ❌ |  **timeouts [^1]** ❌ |
|2.6.0| candle | YES | False | timeouts ✅ | data ✅ | 
|2.6.0| candle | YES | **True** | **hang** ❌ |  **data** ✅ |
|2.6.0| slcan | YES | False | timeouts ✅ | timeouts [^1] ❌ | 
|2.6.0| slcan | YES | **True** | **hang** ❌ | **hang** ❌ |
|this PR| candle | YES | False | timeouts ✅ | data ✅ | 
|this PR| candle | YES | **True** | timeouts ✅ |  data ✅ |
|this PR| slcan | YES | False | timeouts ✅ | data ✅ | 
|this PR| slcan | YES | **True** | timeouts ✅ | data ✅ |

builds on top of #4929 
fixes (completely) #4838

### CODE REVIEW

**Comment by BenGardiner** [comment]
hi @polybassa please consider this PR as a replacement for #4929 since it is commits on-top-of it and completely resolves all issues mentioned in #4838

**Comment by polybassa** [comment]
Luckily we have a lot of unit tests including "stress" tests for the ISOTPSoftSocket. Crossing fingers that it will pass the pipelines!

**Comment by BenGardiner** [comment]
> Luckily we have a lot of unit tests including "stress" tests for the ISOTPSoftSocket. Crossing fingers that it will pass the pipelines!

I think the macos failure is unrelated. And almost all the missing code coverage is error handling edge cases.

**Comment by polybassa** [comment]
Looks great!

File: scapy/contrib/cansocket_python_can.py [hunk 1]
```diff
@@ -161,13 +194,25 @@ def register(self, socket, *args, **kwargs):
             if k in self.pool:
                 t = self.pool[k]
                 t.sockets.append(socket)
-                filters = [s.filters for s in t.sockets
```
Comment (polybassa) [inline scapy/contrib/cansocket_python_can.py:164]: Why are the filters removed entirely?

Comment (BenGardiner) [reply scapy/contrib/cansocket_python_can.py:164]: thank you for pointing that out. This is an unintentional change. the filter merging should still be here in the mux BUT it should be avoided in an slcan case. 

I will make modifications to the PR and test the result manually as well.

**Comment by BenGardiner** [comment]
hmmm ok those pipelines went waaaaay worse than in my test PR here https://github.com/BenGardiner/scapy/pull/2  -- I'm looking into why

**Comment by BenGardiner** [comment]
ok @polybassa .
